### PR TITLE
[ONNXModelLoader] Support for onnx mean operator

### PR DESCRIPTION
**Pull request by psubhasc**
Summary: Support for onnx mean operator

Documentation: https://github.com/onnx/onnx/blob/master/docs/Operators.md#mean

Test Plan: Ninja test and OnnxImporterTest

### CODE REVIEW

**Comment by psubhasc** [comment]
@jfix71

**Comment by jfix71** [comment]
Really sorry for the major delays here, I've been super swamped. I will definitely get to reviewing next week.

File: lib/Importer/ONNXModelLoader.cpp [hunk 1]
```diff
@@ -3207,6 +3207,58 @@ Error ONNXModelLoader::loadCmpLTE(const ONNX_NAMESPACE::NodeProto &op,
...
+    }
+    broadcastShape.insert(broadcastShape.begin(),
+                          *std::max_element(dims.begin(), dims.end()));
+  }
+  return Error::success();
+}
+
+Error ONNXModelLoader::loadMean(const ONNX_NAMESPACE::NodeProto &op,
+                                ArgumentDictionaryTy &dict) {
+  size_t numInputTenosrs = op.input_size();
```
Comment (jfix71) [inline lib/Importer/ONNXModelLoader.cpp:3233]: nit, typo: numInputTenosrs

File: lib/Importer/ONNXModelLoader.cpp [hunk 2]
```diff
@@ -3207,6 +3207,58 @@ Error ONNXModelLoader::loadCmpLTE(const ONNX_NAMESPACE::NodeProto &op,
   return Error::success();
 }
 
+static Error getShapeForBroadcast(llvm::ArrayRef<NodeValue> inputs,
```
Comment (jfix71) [inline lib/Importer/ONNXModelLoader.cpp:3210]: Can you add a doxygen comment explaining what this function does? E.g. `takes a list of NodeValues \p inputs and broadcasts them to a common shape based on the maximum dimension of ...`

**Review by jfix71 — APPROVED** [review]
Few nits, LGTM

File: lib/Importer/ONNXModelLoader.cpp (hunk 2 above)
Comment (psubhasc) [reply lib/Importer/ONNXModelLoader.cpp:3255]: Done.

File: lib/Importer/ONNXModelLoader.cpp [hunk 3]
```diff
@@ -3207,6 +3207,58 @@ Error ONNXModelLoader::loadCmpLTE(const ONNX_NAMESPACE::NodeProto &op,
...
+                                  std::vector<dim_t> &broadcastShape) {
+  std::vector<uint32_t> numDims;
+  for (auto &N : inputs) {
+    numDims.push_back(N.dims().size());
+  }
+  uint32_t outputNumDims = *std::max_element(numDims.begin(), numDims.end());
```
[the remark this answers is not in the document]
Comment (psubhasc) [reply lib/Importer/ONNXModelLoader.cpp]: Done.

File: lib/Importer/ONNXModelLoader.cpp (hunk 1 above)
Comment (psubhasc) [reply lib/Importer/ONNXModelLoader.cpp]: Fixed.

**Review by jfix71 — APPROVED** [review]

**Pull request closed without merging**